pkg/scheduler/predicates.go

```diff
@@ -136,19 +136,18 @@ func CheckPodsExceedingCapacity(pods []*api.Pod, capacity api.ResourceList) (fit
 // PodFitsResources calculates fit based on requested, rather than used resources
 func (r *ResourceFit) PodFitsResources(pod *api.Pod, existingPods []*api.Pod, node string) (bool, error) {
 	podRequest := getResourceRequest(pod)
-	if podRequest.milliCPU == 0 && podRequest.memory == 0 {
-		// no resources requested always fits.
-		return true, nil
-	}
 	info, err := r.info.GetNodeInfo(node)
 	if err != nil {
 		return false, err
 	}
+	if podRequest.milliCPU == 0 && podRequest.memory == 0 {
```

Comment:
You subtly changed the semantics by moving this block, but I don't think it matters.